console.rb

```diff
@@ -12,6 +12,80 @@
 # puts "The first apartment is at #{Apartment.first.address}."
 # puts "It has  #{Apartment.first.tenants.count} tenants."
 
-binding.pry
+# binding.pry
 
-puts "ignore this line, it's just here so the binding.pry above works"
+choice = nil
+while choice != 'q'
+
+  2.times {puts "*" * 70}
+  puts "Welcome to Landlord
+        Options:
+        1) See all apartments
+        2) See all tenants
+        3) See all apartments and associated tenants
+        4) Look up apartment from tenant name
+        q) Exit the program"
+  2.times {puts "*" * 70}
+  puts ""
+  puts "Please enter your selection"
+
+  choice = nil
+  while ![1, 2, 3, 4, 'q'].include? choice
+    choice = (gets.chomp)
+    choice = choice.to_i unless choice == 'q'
+  end
+
+  case choice
+  when 1
+    puts 'All apartments'
+    puts 'ID    | Address                  | Rent   | Sq Feet  | Beds | Baths'
+    puts '*' * 70
+    all_apartments = Apartment.all
+    all_apartments.each do |apartment|
+      print apartment.id, (' ' * (6-apartment.id.to_s.length)), '| ',
+        apartment.address, (' ' * (25-apartment.address.to_s.length)), '| ',
+        apartment.monthly_rent, (' ' * (7-apartment.monthly_rent.to_s.length)), '| ',
+        apartment.sqft, (' ' * (9-apartment.sqft.to_s.length)), '| ',
+        apartment.num_beds, (' ' * (5-apartment.num_beds.to_s.length)), '| ',
+        apartment.num_baths
+      puts ''
+    end
+
+  when 2
+    puts 'All tenants'
+    puts 'ID    | Name                   | Gender         | Age   | Apartment ID'
+    puts '*' * 70
+    all_tenants = Tenant.all
+    all_tenants.each do |tenant|
+      print tenant.id, (' ' * (6-tenant.id.to_s.length)), '| ',
+        tenant.name, (' ' * (23-tenant.name.to_s.length)), '| ',
+        tenant.gender, (' ' * (15-tenant.gender.to_s.length)), '| ',
+        tenant.age, (' ' * (6-tenant.age.to_s.length)), '| ',
+        tenant.apartment_id
+      puts ''
+    end
+
+  when 3
+    puts 'Apartments listed with associated tenants'
+    all_apartments = Apartment.all
+    all_tenants = Tenant.all
+    all_apartments.each do |apartment|
+      print apartment.address
+      puts ''
+      tenants_in_apartment = all_tenants.select {|tenant| tenant.apartment_id == apartment.id}
```

Comment:
Now that you have AR, you can use associations to prevent using select. (at scale, select can be 1000x slower or more, compared to using AR associations).

```ruby
all_apartments.each do |apartment|
  print apartment.address
  puts ' '
  tenants_in_apartment = apartment.tenants
  # ...
```